File: solutions/02_sort_and_search/05.py
n = int(input())
markers = []
for _ in range(n):
  left, r = list(map(int, input().strip().split()))
  markers.extend([(left, 1), (r, -1)])
res, curr = 0, 0
markers.sort()
for m in markers:
  p, typ = m
  curr += typ
  res = max(res, curr)
print(res)

File: solutions/02_sort_and_search/05.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
  int n;
  cin >> n;
  vector<pair<int, int>> markers;
  markers.reserve(2 * n);
  for (int i = 0; i < n; i++) {
    int l, r;
    cin >> l >> r;
    markers.push_back({l, 1});
    markers.push_back({r, -1});
  }
  sort(markers.begin(), markers.end());
  int curr = 0, res = 0;
  for (auto m: markers) {
    auto [_, typ] = m;
    curr += typ;
    res = max(res, curr);
  }
  cout << res << endl;
  return 0;
}